apps/indexer/bin/comment-references-refresher-worker.ts

```diff
@@ -0,0 +1,239 @@
+  const failedResults = await db
+    .select({
+      commentId: schema.commentReferenceResolutionResults.commentId,
+      commentRevision: schema.commentReferenceResolutionResults.commentRevision,
+      references: schema.commentReferenceResolutionResults.references,
+      referencesResolutionStatus:
+        schema.commentReferenceResolutionResults.referencesResolutionStatus,
+      updatedAt: schema.commentReferenceResolutionResults.updatedAt,
+    })
+    .from(schema.commentReferenceResolutionResults)
+    .where(
+      and(
+        inArray(
+          schema.commentReferenceResolutionResults.referencesResolutionStatus,
+          ["failed", "partial"],
+        ),
+        gt(schema.commentReferenceResolutionResults.createdAt, cutoffTime),
+      ),
+    )
+    .limit(options.batchSize);
+
+  if (failedResults.length === 0) {
+    console.log("No failed/partial results found to retry");
+    return;
+  }
+
+  console.log(`Found ${failedResults.length} failed/partial results to retry`);
+
+  if (options.dryRun) {
+    console.log("DRY RUN - Would process the following records:");
+    failedResults.forEach((result, index) => {
+      console.log(
+        `  ${index + 1}. CommentId: ${result.commentId}, Revision: ${result.commentRevision}, Status: ${result.referencesResolutionStatus}, UpdatedAt: ${result.updatedAt.toISOString()}`,
+      );
+    });
+    return;
+  }
+
+  // Process each failed result
+  let successCount = 0;
+  let partialCount = 0;
+  let failedCount = 0;
+
+  for (const result of failedResults) {
+    if (abortController.signal.aborted) {
+      console.log("Abort signal received, stopping processing");
+      break;
+    }
+
+    console.log(
+      `Retrying resolution for commentId: ${result.commentId}, revision: ${result.commentRevision}`,
+    );
+
+    try {
+      // Get the comment details to retry resolution
+      const comment = await db.query.comment.findFirst({
+        where: eq(schema.comment.id, result.commentId),
+      });
+
+      if (!comment) {
+        failedCount++;
+        console.warn(
+          `Comment not found for commentId: ${result.commentId}, skipping`,
+        );
+        continue;
+      }
+
+      const resolved =
+        await commentReferencesResolutionService.resolveFromNetworkFirst({
+          commentId: result.commentId,
+          commentRevision: result.commentRevision,
+          content: comment.content,
+          chainId: comment.chainId,
+        });
+
+      switch (resolved.status) {
+        case "success":
+          successCount++;
+          break;
+        case "partial":
+          partialCount++;
+          break;
+        case "failed":
+          failedCount++;
+          break;
+      }
+
+      await db
+        .update(schema.comment)
+        .set({
+          references: resolved.references,
+          referencesResolutionStatus: resolved.status,
+        })
+        .where(eq(schema.comment.id, result.commentId));
```

Comment:
we should also send an event to webhook if the comment has been updated.

Also you should probably update the `updatedAt` column on the comment